qiskit/transpiler/basepasses.py

```diff
@@ -95,6 +95,45 @@ def is_analysis_pass(self):
         """
         return isinstance(self, AnalysisPass)
 
+    def __call__(self, circuit, property_set=None):
+        """Runs the pass on circuit.
+
+        Args:
+            circuit (QuantumCircuit): the dag on which the pass is run.
+            property_set (PropertySet or dict or None): input/output property set.
+
+        Returns:
+            QuantumCircuit: If on transformation pass, the resulting QuantumCircuit. If analysis
+                   pass, the input circuit.
+        """
+        from qiskit.converters import circuit_to_dag, dag_to_circuit
+        from qiskit.dagcircuit.dagcircuit import DAGCircuit
+
+        property_set_ = None
+        if isinstance(property_set, dict):
+            property_set_ = PropertySet(property_set)
+
+        if isinstance(property_set_, PropertySet):
+            self.property_set = property_set_
+
+        result = self.run(circuit_to_dag(circuit))
+
+        result_circuit = circuit
+
+        if isinstance(property_set, (dict, PropertySet)):
+            property_set.update(self.property_set)
+
+        if isinstance(result, DAGCircuit):
+            result_circuit = dag_to_circuit(result)
+
+        if result is None and self.property_set['layout']:
+            result_circuit = circuit.copy()
+
+        if self.property_set['layout']:
+            result_circuit._layout = self.property_set['layout']
```

Comment:
Should this be included in the `if` block above? Would we ever update `_layout` without copying the circuit?

Reply:
you are right. Done in 567a38cef